tree-sitter/src/TreeSitter/GenerateSyntax.hs

```diff
@@ -68,7 +73,12 @@ syntaxDatatype language datatype = do
       result <- symbolMatchingInstance language name datatypeName
       pure $ generatedDatatype name [con] typeParameterName:result
   where
-    name = toName (datatypeNameStatus datatype) (getDatatypeName (TreeSitter.Deserialize.datatypeName datatype))
+    -- Skip generating datatypes that have already been defined (overridden) in the module where the splice is running.
+    skipDefined m = do
+      isLocal <- lookupTypeName nameStr >>= maybe (pure False) isLocalName
+      if isLocal then pure [] else m
```

Comment:
We only skip definitions from the module where the splice is being run so that we can produce better error messages when someone has e.g. simply forgotten to hide the import of `String` from  `Prelude`—we want the default “ambiguous blah” message, and not for it to try to use `String` (which is of kind `*`) at kind `* -> *` and result in a confusing mismatched kind message.